src/_guides/libraries/library-tour.md

```diff
@@ -351,10 +351,7 @@ items to and removing items from lists.
 
 <?code-excerpt "misc/test/library_tour/core_test.dart (List)"?>
 ```dart
-// Use a List constructor.
-var vegetables = List();
```

Comment:
Same as previous, empty lists are useful.

Reply:
With null safety the empty list constructor is now disallowed.

Reply:
Really? I doubt that. 

A pre-sized list of non optional type is disallowed, as it can’t be pre-filled with nulls. But an empty list of non optional sis still completely valid. And useful.

Reply:
![image](https://user-images.githubusercontent.com/18372958/109434654-f7601500-79db-11eb-9884-bf0371541e8b.png)

Reply:
@parlough is right. There is no `List()` anymore. Only `List.empty()`, and if you want to ever add to it, you have to do `List.empty(growable: true)`.

Reply:
There is also `<type>[]` which has been the idiomatic type constructor for a while.

Reply:
I can create a follow up PR with an example of that.

Reply:
I stand corrected https://dart.dev/tools/diagnostic-messages#default_list_constructor

however, I still advocate for replacing the `List()` constructor with the List literal `[]` instead of deleting it.

Reply:
List literals are covered in the language tour, and there's a link supposedly to that coverage in the text above... but it turns out to be a self-link!

Reply:
I see @parlough has a PR to add back an empty list example: #3017